### PR TITLE
Address Ruff rule RET504 violations

### PR DESCRIPTION
**Pull request by TanviPooranmal**
This pull request is to address Ruff rule ``RET504`` violations in complete ``Astropy`` codebase.

Fixes #14818 ``RET504``.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by TanviPooranmal** [comment]
Draft till I address the failing tests and remaining ``RET504`` violations. Thanks!

File: astropy/cosmology/core.py [hunk 1]
```diff
@@ -602,7 +600,7 @@ def __equiv__(self, other):
 
         # Check if have equivalent parameters and all parameters in `other`
         # match those in `self`` and `other`` has no extra parameters.
-        params_eq = (
+        return = (
```
Comment (mhvk) [inline astropy/cosmology/core.py:603]: Oops! remove `=`!

**Review by mhvk — COMMENTED** [review]
Thanks! To me, these are indeed improvements, except for one place where I would reorder the logic (but fine to leave it as is). But do fix the equal sign that was not removed yet!

**Comment by MridulS** [comment]
@TanviPooranmal a quick way to make all this changes is to remove `RET504` from `.ruff.toml` and then run `ruff check --unsafe-fixes --fix`, also it's recommended to not push every single commit as this will overwhelm the github actions CI servers :)

**Comment by TanviPooranmal** [comment]
pre-commit.ci

**Comment by pllim** [comment]
97 commits? 😱

**Comment by TanviPooranmal** [comment]
I didn't mean to have these many commits, I am still learning. Apologies!

**Comment by neutrinoceros** [comment]
@TanviPooranmal don't worry it won't be a problem, we'll squash your branch into a single commit when it's merged !

**Comment by TanviPooranmal** [comment]
Thank you!

**Comment by pllim** [comment]
Commit one by one via GitHub UI is convenient but only practical for a simple trivial fix. For something that touches so many files like this, I recommend you read the https://docs.astropy.org/en/latest/development/workflow/development_workflow.html (and any other relevant pages), clone your fork properly on your machine, make a feature branch locally, do all the changes there, do local testing and so on, before making a pull request.

Especially given the CI is failing, I think this PR is too messy to go forward. We will be happy to accept your contribution from a cleaner branch. Sorry about that.

**Comment by TanviPooranmal** [comment]
@pllim - Got your point! I'll keep that in mind from now onwards.

**Pull request closed without merging**